SpotifyExtractor.py

```diff
@@ -0,0 +1,173 @@
+            return self.access_token
+
+        auth_string = f"{self.CLIENT_ID}:{self.CLIENT_SECRET}"
+        auth_bytes = auth_string.encode('utf-8')
+        auth_base64 = base64.b64encode(auth_bytes).decode('utf-8')
+
+        headers = {
+            'Authorization': f'Basic {auth_base64}',
+            'Content-Type': 'application/x-www-form-urlencoded'
+        }
+        data = {'grant_type': 'client_credentials'}
+
+        try:
+            response = requests.post('https://accounts.spotify.com/api/token',
+                                     headers=headers,
+                                     data=data,
+                                     timeout=10)
+            response.raise_for_status()
+            token_data = response.json()
+            self.access_token = token_data['access_token']
+            self.token_expiry = datetime.now() + timedelta(seconds=token_data['expires_in'] - 60)
+            return self.access_token
+        except Exception as e:
+            logging.error(f"Error getting access token: {str(e)}")
+            return None
+
+    def make_request(self, endpoint: str) -> Optional[Dict]:
+        headers = {'Authorization': f'Bearer {self.get_access_token()}'}
+
+        try:
+            response = requests.get(f'https://api.spotify.com/v1/{endpoint}',
+                                    headers=headers,
+                                    timeout=15)
+            if response.status_code == 200:
+                return response.json()
+            elif response.status_code == 401:
+                self.access_token = None
+                return self.make_request(endpoint)
+            else:
+                logging.error(f"API error {response.status_code} for {endpoint}")
+                return None
+        except Exception as e:
+            logging.error(f"Request failed: {str(e)}")
+            return None
+
+    def get_playlist_info(self, playlist_id: str) -> Optional[Dict]:
+        data = self.make_request(f'playlists/{playlist_id}')
+        if not data:
+            return None
+
+        try:
+            playlist_info = {
+                'playlist_id': data['id'],
+                'playlist_name': data['name'],
+                'owner_id': data['owner']['id'],
+                'total_tracks': data['tracks']['total'],
+                'public': bool(data['public']),
+                'playlist_uri': data['uri']
+        }
+
+            playlist_info['tracks'] = []
+
+            for track_item in data['tracks']['items']:
+                track_info = {
+                    'added_at': track_item['added_at'],
+                    'track_id': track_item['track']['id'],
+                    'track_name': track_item['track']['name'],
+                }
+                playlist_info['tracks'].append(track_info)
+
+            return playlist_info
+        except KeyError as e:
+            logging.error(f"Missing field in playlist data: {str(e)}")
+            return None
```

Comment:
Is it possible to split this method into 2 separate methods? One to get playlist data, and another one to get playlist_track data. It will be much cleaner